codegen/src/test/kotlin/software/amazon/smithy/rust/tool/TimeTestSuiteGenerator.kt

```diff
@@ -142,35 +145,79 @@ private fun generateDateTimeTests(parsing: Boolean): List<TestCase> {
-    tests["parse_http_date"] = generateHttpDateTests(parsing = true)
-        .map(TestCase::toSerializable)
-        .filter { it.formatted != null }
-    tests["parse_date_time"] = generateDateTimeTests(parsing = true)
-        .map(TestCase::toSerializable)
-        .filter { it.formatted != null }
-
-    println("{")
-    val testSuites = tests.entries.map { entry ->
-        var result = "  \"${entry.key}\": [\n"
-        val testCases = entry.value.map { testCase ->
-            val iso = testCase.iso.replace("\"", "\\\"")
-            val formatted = testCase.formatted?.let { "\"" + it.replace("\"", "\\\"") + "\"" } ?: "null"
-            val secs = testCase.epochSeconds
-            val subsecs = testCase.epochSubsecondNanos
-            """    { "epoch_seconds": $secs, "epoch_subsecond_nanos": $subsecs, "iso": "$iso",
-              |      "formatted": $formatted }""".trimMargin()
-        }
-        result += testCases.joinToString(",\n")
-        result += "\n  ]"
-        result
-    }
-    println(testSuites.joinToString(",\n"))
-    println("}")
+    val none = SourceLocation.NONE
+    val topLevels = mapOf<String, Node>(
+        "description" to ArrayNode(
+            """
+            This file holds format and parse test cases for Smithy's built-in `epoch-seconds`,
+            `http-date`, and `date-time` timestamp formats.
+
+            There are six top-level sections:
+             - `format_epoch_seconds`: Test cases for formatting timestamps into `epoch-seconds`
+             - `format_http_date`: Test cases for formatting timestamps into `http-date`
+             - `format_date_time`: Test cases for formatting timestamps into `date-time`
+             - `parse_epoch_seconds`: Test cases for parsing timestamps from `epoch-seconds`
+             - `parse_http_date`: Test cases for parsing timestamps from `http-date`
+             - `parse_date_time`: Test cases for parsing timestamps from `date-time`
+
+            Each top-level section is an array of the same test case data structure:
+            ```typescript
+            type TestCase = {
+                // Human-readable ISO-8601 representation of the canonical date-time. This should not
+                // be used by tests, and is only present to make test failures more human readable.
+                iso8601: string,
+
+                // The canonical number of seconds since the Unix epoch in UTC.
+                canonical_seconds: string,
+
+                // The canonical nanosecond adjustment to the canonical number of seconds.
+                // If conversion from (canonical_seconds, canonical_nanos) into a 128-bit integer is required,
+                // DO NOT just add the two together as this will yield an incorrect value when
+                // canonical_seconds is negative.
+                canonical_nanos: number,
+
+                // Will be true if this test case is expected to result in an error or exception
+                error: boolean,
```

Comment:
don't know if it's worth it, but could make this a string and have a message about why it's an error if it's ever unclear